pkg/jobmanager/start.go

```diff
@@ -16,7 +17,28 @@ import (
 
 func (jm *JobManager) start(ev *api.Event) *api.EventResponse {
 	msg := ev.Msg.(api.EventStartMsg)
-	j, err := NewJob(jm.pluginRegistry, msg.JobDescriptor)
+	var jd job.JobDescriptor
+	if err := json.Unmarshal([]byte(msg.JobDescriptor), &jd); err != nil {
+		return &api.EventResponse{Err: err}
+	}
+	// Add instance tag, if specified.
+	if jm.config.instanceTag != "" {
+		found := false
+		for _, tag := range jd.Tags {
+			if tag == jm.config.instanceTag {
+				found = true
+				break
+			}
+		}
+		if !found {
+			jd.Tags = append(jd.Tags, jm.config.instanceTag)
+		}
```

Comment:
will this explode if a user submits a job with a tag that is also used as an instance tag of some other instance?

Reply:
it won't immediately explode and in fact, at present, nothing at all will happen. however, in the future, their paused job may be resumed by another instance which will probably not go well.